src/state/pools/fetchVaultUser.ts

```diff
@@ -0,0 +1,27 @@
+import BigNumber from 'bignumber.js'
+import { getCakeVaultContract } from 'utils/contractHelpers'
+
+const cakeVaultContract = getCakeVaultContract()
+
+const fetchVaultUser = async (account: string) => {
+  try {
+    const userContractResponse = await cakeVaultContract.methods.userInfo(account).call()
+    return {
+      isLoading: false,
+      userShares: new BigNumber(userContractResponse.shares as string).toJSON(),
```

Comment:
Is this a redundant exercise? Could I just store this contract response as-is, as a string, and then convert to a BigNumber when retrieving from state? (I have hooks doing this second part already - https://github.com/ChefHutch/pancake-frontend/pull/11/files#r628241402)

Reply:
is `as string` necessary?

Reply:
Ah, no it probably isn't for these BigNumbers @hachiojidev - I'll remove.

On the serialization, I've had a few of these warnings thrown in the console - which I think is relating to this? Any idea if it's something to be concerned about?

```
utils.ts:17 SerializableStateInvariantMiddleware took 157ms, which is more than the warning threshold of 32ms. 
If your state or actions are very large, you may want to disable the middleware as it might cause too much of a slowdown in development mode. See https://redux-toolkit.js.org/api/getDefaultMiddleware for instructions.
It is disabled in production builds, so you don't need to worry about that.
```